mirgecom/profiling.py

```diff
@@ -101,7 +108,9 @@ def __init__(self, queue, allocator=None, logmgr: LogManager = None) -> None:
         self.kernel_stats = {}
         self.logmgr = logmgr
 
-        cl.array.ARRAY_KERNEL_EXEC_HOOK = self.array_kernel_exec_hook
+        # Only store the first kernel exec hook for elwise kernels
+        if cl.array.ARRAY_KERNEL_EXEC_HOOK is None:
+            cl.array.ARRAY_KERNEL_EXEC_HOOK = self.array_kernel_exec_hook
 
     def clone(self):
         """Return a semantically equivalent but distinct version of *self*."""
```

Comment:
Probably a warning here, that cloned instances have spurious stats collection?

Reply:
Done in https://github.com/illinois-ceesd/mirgecom/pull/476/commits/8ce79338807af61111f5933156165860a037a92d

Reply:
Thanks, good to go form my end.